logic/src/commonMain/kotlin/com/wire/kalium/logic/configuration/UserConfigRepository.kt

```diff
@@ -20,81 +20,104 @@ package com.wire.kalium.logic.configuration
+import kotlinx.coroutines.flow.flowOf
 import kotlinx.coroutines.flow.map
 
 interface UserConfigRepository {
-    fun setFileSharingStatus(status: Boolean, isStatusChanged: Boolean?): Either<StorageFailure, Unit>
-    fun isFileSharingEnabled(): Either<StorageFailure, FileSharingStatus>
-    fun isFileSharingEnabledFlow(): Flow<Either<StorageFailure, FileSharingStatus>>
-    fun setClassifiedDomainsStatus(enabled: Boolean, domains: List<String>): Either<StorageFailure, Unit>
-    fun getClassifiedDomainsStatus(): Flow<Either<StorageFailure, ClassifiedDomainsStatus>>
-    fun isMLSEnabled(): Either<StorageFailure, Boolean>
-    fun setMLSEnabled(enabled: Boolean): Either<StorageFailure, Unit>
-    fun setConferenceCallingEnabled(enabled: Boolean): Either<StorageFailure, Unit>
-    fun isConferenceCallingEnabled(): Either<StorageFailure, Boolean>
-    fun isReadReceiptsEnabled(): Flow<Either<StorageFailure, Boolean>>
-    fun setReadReceiptsStatus(enabled: Boolean): Either<StorageFailure, Unit>
+    suspend fun setFileSharingStatus(status: Boolean, isStatusChanged: Boolean?): Either<StorageFailure, Unit>
+    suspend fun isFileSharingEnabled(): Either<StorageFailure, FileSharingStatus>
+    suspend fun isFileSharingEnabledFlow(): Flow<Either<StorageFailure, FileSharingStatus>>
+    suspend fun setClassifiedDomainsStatus(enabled: Boolean, domains: List<String>): Either<StorageFailure, Unit>
+    suspend fun getClassifiedDomainsStatus(): Flow<Either<StorageFailure, ClassifiedDomainsStatus>>
+    suspend fun isMLSEnabled(): Either<StorageFailure, Boolean>
+    suspend fun setMLSEnabled(enabled: Boolean): Either<StorageFailure, Unit>
+    suspend fun setConferenceCallingEnabled(enabled: Boolean): Either<StorageFailure, Unit>
+    suspend fun isConferenceCallingEnabled(): Either<StorageFailure, Boolean>
+    suspend fun isReadReceiptsEnabled(): Flow<Either<StorageFailure, Boolean>>
+    suspend fun setReadReceiptsStatus(enabled: Boolean): Either<StorageFailure, Unit>
 }
 
 class UserConfigDataSource(
-    private val userConfigStorage: UserConfigStorage
+//     private val userConfigStorage: UserConfigStorage,
+    private val metadataDAO: MetadataDAO
 ) : UserConfigRepository {
-
-    override fun setFileSharingStatus(status: Boolean, isStatusChanged: Boolean?): Either<StorageFailure, Unit> =
-        wrapStorageRequest { userConfigStorage.persistFileSharingStatus(status, isStatusChanged) }
-
-    override fun isFileSharingEnabled(): Either<StorageFailure, FileSharingStatus> =
-        wrapStorageRequest { userConfigStorage.isFileSharingEnabled() }.map {
-            with(it) { FileSharingStatus(status, isStatusChanged) }
+    override suspend fun setFileSharingStatus(status: Boolean, isStatusChanged: Boolean?): Either<StorageFailure, Unit> =
+        wrapStorageRequest {
+            metadataDAO.insertValue(
+                JsonSerializer().encodeToString(FileSharingStatus.serializer(), FileSharingStatus(status, isStatusChanged)),
```

Comment:
No need to do to pass the serializer you can use JsonSerializer().encodeToString(FileSharingStatus(status, isStatusChanged))

or even better add a method in the MetaDataDAO to insert a json  something similar to the function putSerializable is KaliumPreferences